bitwarden_license/test/Scim.Test/Commands/Groups/PutGroupCommandTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Scim.Commands.Groups;
+using Bit.Scim.Context;
+using Bit.Scim.Models;
+using Bit.Scim.Utilities;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using Bit.Test.Common.Helpers;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Scim.Test.Commands.Groups;
+
+[SutProviderCustomize]
+public class PutGroupCommandTests
+{
+    [Theory]
+    [BitAutoData]
+    public async Task PutGroup_Success(SutProvider<PutGroupCommand> sutProvider, Group group, string displayName)
+    {
+        var scimGroupRequestModel = new ScimGroupRequestModel
+        {
+            DisplayName = displayName,
+            Schemas = new List<string> { ScimConstants.Scim2SchemaUser }
+        };
+
+        sutProvider.GetDependency<IGroupRepository>()
+            .GetByIdAsync(group.Id)
+            .Returns(group);
+
+        var expectedResult = new ScimGroupResponseModel(group);
+        expectedResult.DisplayName = displayName;
+
+        var result = await sutProvider.Sut.PutGroupAsync(group.OrganizationId, group.Id, scimGroupRequestModel);
+
+        await sutProvider.GetDependency<IGroupService>().Received(1).SaveAsync(group);
```

Comment:
Should we check that it's actually updated `group.Name`?

Reply:
I've added an extra check for `group.Name` on both the `SaveAsync` group parameter and the returned command result.

Reply:
I was thinking of checking that the group object has the name updated _at the time it's passed_ to `GroupService.SaveAsync`. But because we're mutating the object, this is actually fairly difficult to test in this level of detail. So that's okay, we have integration tests for db state anyway.

Reply:
That is already being tested here:
```
Assert.Equal(displayName, group.Name);

await sutProvider.GetDependency<IGroupService>().Received(1).SaveAsync(group);
```

Reply:
`Received` only checks the object reference, it doesn't do a deep comparison on the object. So we know that the `group` object was passed to `SaveAsync`, but we don't know that it had the correct `Name` at the time. (You can reorder the code in the command so that the name is changed after the `GroupService.SaveAsync` call, this will still pass.)

I thought that maybe we could do `Arg.Is(g => g.Name == displayName)`, but that suffers from the same problem. It compares `g.Name` at the time the matcher is evaluated, not at the time the service was called. See "Modifying values being matched" here: https://nsubstitute.github.io/help/argument-matchers/

But, as I said, not that concerned about it.

Reply:
That's a really good explanation, now I've learned something new. Thanks for that! I'll keep the direct assertion for now.